tests/ui/question_mark.rs

```diff
@@ -186,6 +186,43 @@ fn f() -> NotOption {
     NotOption::First
 }
 
+#[allow(dead_code)]
+mod issue8288 {
+    struct CustomError;
+
+    fn foo() -> Result<u8, CustomError> {
+        Ok(1)
+    }
+
+    /// Should just replace the if let pattern with `foo()?;`
+    fn bar(some_flag: bool) -> Result<u8, CustomError> {
+        if let Err(err) = foo() {
+            return Err(err);
+        }
+        let mut res = 1_u8;
+        if some_flag {
+            res = 0_u8;
+        }
+        Ok(res)
+    }
+
+    /// No other function logic, same result, replace whole function body with `foo()`
+    fn baz() -> Result<u8, CustomError> {
+        if let Err(err) = foo() {
+            return Err(err);
+        }
+        Ok(1)
+    }
+
+    /// No other function logic, different result, don't replace whole function body
+    fn qux() -> Result<u8, CustomError> {
+        if let Err(err) = foo() {
+            return Err(err);
+        }
+        Ok(2)
+    }
+}
```

Comment:
nit: Let's also add a test case for when the `if let` does not immediately return `Err` i.e.

```rust
if let Err(err) = foo() {
    do_something();
    return Err(err);
}
```

Reply:
And this one shouldn't trigger right?